src/compile/selection/index.ts

```diff
@@ -29,12 +30,10 @@ export interface SelectionComponent<T extends SelectionType = SelectionType> {
   name: string;
   type: T;
 
-  // Use conditional typing (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html)
-  // so we have stricter type of init (as the type of init depends on selection type)
   init?: (T extends 'interval'
-    ? SelectionInitInterval
+    ? IntervalSelection['init']['a']
```

Comment:
This seems weird. Why isn't `SelectionInitInterval` working?

Reply:
This get's the right type so when we change `IntervalSelection.init`, the types will be updated correctly.